app/models/miq_request_workflow.rb

```diff
@@ -999,7 +990,7 @@ def add_target(dialog_key, key, klass, result)
     key_id = "#{key}_id".to_sym
     result[key_id] = get_value(@values[dialog_key])
     result[key_id] = nil if result[key_id] == 0
-    result[key] = ci_to_hash_struct(klass.find_by_id(result[key_id])) unless result[key_id].nil?
+    result[key] = ci_to_hash_struct(klass.find(result[key_id])) unless result[key_id].nil?
```

Comment:
This is quite different... find_by_id will return nil if not found; find with raise an exception.